### PR TITLE
Adds basic support for http2 spdy

### LINKED ISSUE #130: http2/spdy
https://github.com/molnarg/node-http2

### PR DESCRIPTION
**Pull request by designfrontier**
First rough run at http2 support. Right now you have to pass it
an http2 server factory like the `http2` module on npm. Which is
what I am using for testing at this point.

This should probably become a flag in config. It also requires
that you pass it a key and cert for it to work.

Not sure exactly how this plays with heroku or other hosted systems

closes #130

### CODE REVIEW

**Comment by designfrontier** [comment]
This also lays the groundwork for https setup as well.

@uniqname http2 is working with this PR. Once it merges it will be ready for primetime. If you have a chance to play with it that would be excellent

**Comment by designfrontier** [comment]
The big question is API related... Do we activate this through a flag? Or do we allow you to pass in any server that you want (http, https, http2) and then go from there?

I am leaning to the more flexible of these options in allowing you to pass in a server type as part of the config options. Seems like it is the most future friendly and the most flexible.

**Comment by designfrontier** [comment]
To use this branch: 
1) create a cert and key in the base of our project https://devcenter.heroku.com/articles/ssl-certificate-self
2) `npm install --save http2`
3) pass the cert as config.cert, the key as config.key, and the server as config.server
4) fire up the server with `npm start`
5) hit it over `https://localhost:3000` and make sure it is working
6) connection.res.push should now be available when the browser supports http2. This is how you push additional files to the client on the same connection and it explained more in the [http2](https://github.com/molnarg/node-http2) documentation.

**Comment by uniqname** [comment]
Cool! I'll take a look. I think I lean the same way you do on the configuration.

File: routes/index.js [hunk 1]
```diff
@@ -7,7 +7,17 @@ const parseRoutes = require('./parseRoutes')
 
 
     , server = (serverType, routesJson, config) => {
-        return serverType.createServer(require('./router.js')(routesJson, config));
+        let serverReturn;
+
+        if (config.key && config.cert) {
+            serverReturn = serverType.createServer({
```
Comment (uniqname) [inline routes/index.js]: Is it needful to document the interface the httpServer should have or is there sufficient convention around Node httpServers (http/https,http2, etc) to have a `createServer` method for, uh, creating the server :)

Comment (designfrontier) [inline routes/index.js]: The convention is pretty solid. All the libraries I've found for http2/spdy is a createServer function with an options object as the first parameter. Which is the same signature and function used by the built in http/https modules.

Based on that I think that documentation might be nice but is not a must have.

-Daniel

**Comment by designfrontier** [comment]
The documentation now points to using `spdy` since it seems to be the most common http2/spdy module out there. It does require a `ca` cert as well as the normal key/cert combo. Haven't had time to try it yet... but because its API is the same as everything else it shouldn't change anything on the `monument` side of the world

**Pull request merged**